src/tls/openssl.c

```diff
@@ -434,6 +434,13 @@ static int tls_net_read(struct flb_tls_session *session,
             ERR_error_string_n(ret, err_buf, sizeof(err_buf)-1);
             flb_error("[tls] syscall error: %s", err_buf);
 
+            /* According to the documentation these are non-recoverable
+             * errors so we don't need to screen them before saving them
+             * to the net_error field.
+             */
+
+            session->connection->net_error = errno;
```

Comment:
Just wondering, should the async paths be covered as well? https://github.com/fluent/fluent-bit/blob/072439b80f00e215dd4f889651fd47d1e8205f70/src/flb_io.c#L656

Reply:
None of the `flb_tls_` functions handle it in `flb_tls.c` because `tls_net_read` and `tls_net_write` in `openssl.c` handle it through the `SSL_ERROR_SYSCALL` code paths.